src/overlays/actors/ovl_Bg_Menkuri_Eye/z_bg_menkuri_eye.c

```diff
@@ -27,11 +26,113 @@ const ActorInit Bg_Menkuri_Eye_InitVars = {
     (ActorFunc)BgMenkuriEye_Update,
     (ActorFunc)BgMenkuriEye_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Menkuri_Eye/BgMenkuriEye_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Menkuri_Eye/BgMenkuriEye_Destroy.s")
+extern Gfx D_06002D20[];
+UNK_TYPE D_8089C1A0;
+
+static ColliderJntSphItemInit sJntSphItemsInit[1] = {
+    {
+        { 0x04, { 0x00000000, 0x00, 0x00 }, { 0x0001F820, 0x00, 0x00 }, 0x00, 0x01, 0x00 },
+        { 1, { { 0, 0, 0 }, 14 }, 100 },
+    },
+};
+
+static ColliderJntSphInit sJntSphInit = {
+    { COLTYPE_UNK10, 0x00, 0x09, 0x00, 0x20, COLSHAPE_JNTSPH },
+    1,
+    sJntSphItemsInit,
+};
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
+};
+
+void BgMenkuriEye_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgMenkuriEye* this = THIS;
+    ColliderJntSphItem* colliderList;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    Collider_InitJntSph(globalCtx, &this->collider);
+    Collider_SetJntSph(globalCtx, &this->collider, &this->actor, &sJntSphInit, &this->colliderItem);
+    this->collider.list->dim.worldSphere.center.x = this->actor.posRot.pos.x;
+    this->collider.list->dim.worldSphere.center.y = this->actor.posRot.pos.y;
+    this->collider.list->dim.worldSphere.center.z = this->actor.posRot.pos.z;
+    colliderList = this->collider.list;
+    colliderList->dim.worldSphere.radius = colliderList->dim.modelSphere.radius;
+    if (!Flags_GetSwitch(globalCtx, this->actor.params)) {
+        D_8089C1A0 = 0;
+    }
+    this->unk_14C = -1;
+}
+
+void BgMenkuriEye_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgMenkuriEye* this = THIS;
+
+    Collider_DestroyJntSph(globalCtx, &this->collider);
+}
+
+void BgMenkuriEye_Update(Actor* thisx, GlobalContext* globalCtx) {
+    BgMenkuriEye* this = THIS;
+    s32 temp;
+
+    if (!Flags_GetSwitch(globalCtx, this->actor.params)) {
+        if (this->unk_14C != -1) {
+            if (this->unk_14C != 0) {
+                this->unk_14C -= 1;
+            }
+            if (this->unk_14C == 0) {
+                this->unk_14C = -1;
+                D_8089C1A0 -= 1;
+            }
+        }
+    }
+    if (this->collider.base.acFlags & 2) {
+        temp = ABS((s16)(this->collider.base.ac->posRot.rot.y - this->actor.shape.rot.y));
+        if (temp > 0x5000) {
+            this->collider.base.acFlags &= ~0x2;
+            if (this->unk_14C == -1) {
+                Audio_PlayActorSound2(&this->actor, NA_SE_EN_AMOS_DAMAGE);
+                D_8089C1A0 += 1;
+                if (D_8089C1A0 > 4) {
+                    D_8089C1A0 = 4;
+                } else {
+                    D_8089C1A0 = D_8089C1A0;
+                }
+            }
+            this->unk_14C = 0x1A0;
```

Comment:
`unk_14C` looks like a timer so maybe frame amount in base 10 (decimal) makes more sense?